### PR TITLE
Ft integration

### PR DESCRIPTION
**Pull request by bright-starry-sky**
integrat PR #365 and #395 .  for Integration testing.

### CODE REVIEW

**Comment by bright-starry-sky** [comment]
Resolved conflict
Refer to https://github.com/vesoft-inc/nebula-common/pull/260

**Comment by bright-starry-sky** [comment]
Resolved conflict

File: src/validator/IndexScanValidator.cpp [hunk 1]
```diff
@@ -93,26 +95,147 @@ Status IndexScanValidator::prepareFilter() {
...
+        auto retExpr = rewriteTSFilter(filter);
+        if (!retExpr.ok()) {
+            return retExpr.status();
+        }
+        if (isEmptyResultSet_) {
+            // return empty result direct.
+            return Status::OK();
+        }
+        auto ret = checkFilter(retExpr.value());
+        NG_RETURN_IF_ERROR(ret);
```
Comment (yixinglu) [inline src/validator/IndexScanValidator.cpp:113]: Leak retExpr.value() memory when fail to checkFilter

File: src/util/ExpressionUtils.cpp [hunk 2]
```diff
@@ -75,5 +75,14 @@ std::unique_ptr<InputPropertyExpression> ExpressionUtils::inputPropExpr(const st
     return std::make_unique<InputPropertyExpression>(new std::string(prop));
 }
 
+Expression*
+ExpressionUtils::pushOrs(std::vector<RelationalExpression*> rels) {
```
[the remark this answers is not in the document]
Comment (bright-starry-sky) [reply src/util/ExpressionUtils.cpp]: agree

File: src/validator/IndexScanValidator.cpp (hunk 1 above)
Comment (bright-starry-sky) [reply src/validator/IndexScanValidator.cpp]: Good point, fixed . thanks.

**Review by bright-starry-sky — COMMENTED** [review]
Addressed yee's comments.

**Review by yixinglu — APPROVED** [review]

**Review by critical27 — APPROVED** [review]
LGTM

**Review by dutor — APPROVED** [review]

File: src/validator/IndexScanValidator.cpp [hunk 3]
```diff
@@ -93,26 +95,157 @@ Status IndexScanValidator::prepareFilter() {
...
+        NG_RETURN_IF_ERROR(checkTSService());
+        if (!textSearchReady_) {
+            return Status::Error("Text search service not ready");
+        }
+        auto retFilter = rewriteTSFilter(filter);
+        if (!retFilter.ok()) {
+            return retFilter.status();
+        }
+        if (isEmptyResultSet_) {
+            // return empty result direct.
```
Comment (Amber1990Zhang) [inline src/validator/IndexScanValidator.cpp:109]: return empty result directly

**Pull request merged**